src/main/java/com/airdnb/clone/domain/reservation/request/ReservationSaveRequest.java

```diff
@@ -0,0 +1,43 @@
+package com.airdnb.clone.domain.reservation.request;
+
+import com.airdnb.clone.domain.common.Guest;
+import com.airdnb.clone.domain.reservation.entity.Reservation;
+import com.airdnb.clone.domain.reservation.entity.Reservation.ReservationBuilder;
+import jakarta.validation.constraints.Future;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
+public class ReservationSaveRequest {
+
+    @NotNull
+    private final Long memberId;
+
+    @NotNull
+    private final Long stayId;
+
+    @Future
+    private final LocalDateTime checkIn;
+
+    @Future
+    private final LocalDateTime checkOut;
+
+    @Min(value = 0)
+    private final Long totalRate;
+
+    @Min(value = 1)
+    private final Integer guestCount;
+
+    public ReservationBuilder toBuilder() {
+        return Reservation.builder()
+                .checkIn(checkIn)
+                .checkOut(checkOut)
+                .totalRate(totalRate)
+                .guest(Guest.builder().guestCount(guestCount).build());
```

Comment:
* suggestion 코드처럼 Stay 엔티티와 Member 엔티티를 파라미터로 받아서 예약 엔티티를 반환하지 않은 이유는
    * 예약Dto 가 Stay 와 Member 를 의존 할 필요가 없다 생각되어 위 코드처럼 작성하게 되었습니다.
* 다만, Entitiy의 Builder 를 반환하는 패턴이 익숙치가 않아서 자주 사용되는지, 아니라면 왜 자주 사용되지 않는지 궁금한지 궁금합니다!
```suggestion
    public Reservation toEntity(Stay stay, Member member) {
        return Reservation.builder()
                .stay(stay)
                .member(member)
                .checkIn(checkIn)
                .checkOut(checkOut)
                .totalRate(totalRate)
                .guest(Guest.builder().guestCount(guestCount).build())
                .build();
```
* toBuilder 를 호출하는 [서비스( `create` 메서드)](https://github.com/codesquad-members-2024/be-airdnb/pull/37/files#diff-dbfe21a969569d6694c476ab18db9f41bbf22cff26bc587b65a85ee1cb0f2a70) 입니다.

Reply:
음, 질문의 의도를 살짝 이해하기 어려운데요. Builder 가 자주 쓰인다고 판단하신 자료들도 함께 보여주셔도 좋을거 같아요. 사실 자주쓰이고 이런 것보다, 제 생각에는 의도가 더 중요할거 같아요. 메소드를 제공하는 입장에서 생각해보면, Builder 를 반환할때는 보통 Builder 를 통해 더 building 이 필요한 경우 (추가로 값을 받아야 할 경우) 가 아니라면 굳이 Builder 로 반환할 필요는 없을거 같아요. 
의도치 않은 값 변환 혹은 추가가 일어날 수도 있고 의도가 모호해지는거 같아요.